core/trino-main/src/main/java/io/trino/server/remotetask/DynamicFiltersFetcher.java

```diff
@@ -163,26 +166,43 @@ private synchronized void fetchDynamicFiltersIfNecessary()
 
         errorTracker.startRequest();
         future = httpClient.executeAsync(request, createFullJsonResponseHandler(dynamicFilterDomainsCodec));
-        addCallback(future, new SimpleHttpResponseHandler<>(new DynamicFiltersResponseCallback(), request.getUri(), stats), executor);
+        addCallback(future, new SimpleHttpResponseHandler<>(new DynamicFiltersResponseCallback(dynamicFiltersVersion), request.getUri(), stats), executor);
     }
 
     private class DynamicFiltersResponseCallback
             implements SimpleHttpResponseCallback<VersionedDynamicFilterDomains>
     {
         private final long requestStartNanos = System.nanoTime();
+        private final long requestedDynamicFiltersVersion;
+
+        public DynamicFiltersResponseCallback(long requestedDynamicFiltersVersion)
+        {
+            this.requestedDynamicFiltersVersion = requestedDynamicFiltersVersion;
+        }
 
         @Override
         public void success(VersionedDynamicFilterDomains newDynamicFilterDomains)
         {
             try (SetThreadName ignored = new SetThreadName("DynamicFiltersFetcher-%s", taskId)) {
                 updateStats(requestStartNanos);
                 try {
-                    updateDynamicFilterDomains(newDynamicFilterDomains);
-                    errorTracker.requestSucceeded();
+                    if (newDynamicFilterDomains.getVersion() < requestedDynamicFiltersVersion) {
+                        // Receiving older dynamic filter shouldn't happen unless
```

Comment:
What if coordinator and worker are both at initial version of 0 ?
In that case, we wouldn't be able to use this logic right ?
To me it feels like the worker should somehow generically reject all task requests that are not meant for it. Not sure how feasible that is.

Reply:
> In that case, we wouldn't be able to use this logic right ?

That is fine as 0 means `all` tuple domain (it's initial state). Also once coordinator sees the worker version, it will stop pinging worker, which was an issue previously